README.md

```diff
@@ -23,21 +23,46 @@ $ pip install pip-tools
 Example usage for `pip-compile`
 ===============================
 
-Suppose you have a Flask project, and want to pin it for production.  Write the
-following line to a file:
+Requirements from setup.py
+--------------------------
+
+Suppose you have a Flask project, and want to pin it for production.  If you have a `setup.py` with
+`install_requires=['Flask']`, then run `pip-compile` without any arguments:
+```console
+$ pip-compile
+#
+# This file is autogenerated by pip-compile
+# To update, run:
+#
+#    pip-compile --output-file requirements.txt setup.py
+#
+flask==0.10.1
+itsdangerous==0.24        # via flask
+jinja2==2.7.3             # via flask
+markupsafe==0.23          # via jinja2
+werkzeug==0.10.4          # via flask
+```
+
+`pip-compile` will produce your `requirements.txt`, with all the Flask dependencies
+(and all underlying dependencies) pinned.  You should put your requirements file under version control.
+
+Without setup.py
+----------------
+
+If you don't use `setup.py` ([you should][1]), you can write the following line to a file:
```

Comment:
>you should

This is an opinion [not everyone would agree with](https://caremad.io/posts/2013/07/setup-vs-requirement/).